app/src/Components/InteractionBar.js

```diff
@@ -1,31 +1,42 @@
-import React from "react";
+import React, { useState } from "react";
 import "./InteractionBar.css";
 
 const InteractionBar = () => {
+  const [commentCount, setCommentCount] = useState(1);
+  const [retweetCount, setRetweetCount] = useState(5);
+  const [likeCount, setLikeCount] = useState(100);
+  const [shareCount, setShareCount] = useState(10);
+
   return (
     <div className="interactionIcons">
-
+      
       <div className="comment">
-        <i className="ri-chat-3-line"> </i>
-        <p>50</p>
+        <button type="button" onClick={() => setCommentCount(commentCount + 1)}>
+          <i className="ri-chat-3-line"> </i>
+        </button>
+        <p> {commentCount} </p>
       </div>
 
       <div className="retweet">
-        <i className="ri-refresh-line"> </i>
-        <p>10</p>
+        <button type="button" onClick={() => setRetweetCount(retweetCount + 1)}>
+          <i className="ri-refresh-line"> </i>
+        </button>
+        <p> {retweetCount} </p>
       </div>
 
       <div className="like">
-        <i className="ri-heart-line"> </i>
-        {/* <i class="ri-heart-fill">100</i> when reg heart clicked this should be triggered */}
-        <p>123</p>
+        <button type="button" onClick={() => setLikeCount(likeCount + 1)}>
+          <i className="ri-heart-line"> </i>
+        </button>
+        <p> {likeCount} </p>
       </div>
-      
+
       <div className="share">
-        <i className="ri-download-2-line"> </i>
-        <p>5</p>
+        <button type="button" onClick={() => setShareCount(shareCount + 1)}>
+          <i className="ri-download-2-line"> </i>
+        </button>
+        <p> {shareCount} </p>
       </div>
-      
     </div>
   );
 };
```

Comment:
Maybe this could be refactored to one icon component that gets reused in the future?

Reply:
Sorry, this is supposed to be a comment on all the lines above (interactionBar return JSX)